gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py

```diff
@@ -0,0 +1,887 @@
+                     "description": '{\
+                                    "vm_management_ip": "172.24.4.5",\
+                                    "service_vendor": "vyos"}',
+                     "firewall_rule_list": True
+                    }
+        return firewall
+
+    def fake_request_data_generic_config(self):
+        data = {
+                'context': {},
+                'kwargs': self._fake_kwargs(),
+                'request_info': 'xxx'}
+
+        request_data = {
+                 'info':   {
+                             'version': 'v1'},
+
+                 'config': [
+                        {'resource': 'interfaces',
+                         'kwargs': data}
+                    ]
+                }
+
+        return request_data
+
+
+class ConfiguratorRpcManagerTestCase(unittest.TestCase):
+    ''' Configurator RPC receiver '''
+
+    def __init__(self, *args, **kwargs):
+        super(ConfiguratorRpcManagerTestCase, self).__init__(*args, **kwargs)
+        self.fo = FakeObjects()
+
+    @mock.patch(__name__ + '.FakeObjects.conf')
+    @mock.patch(__name__ + '.FakeObjects.sc')
+    def _get_ConfiguratorRpcManager_object(self,sc, conf):
+        cm = cfgr.ConfiguratorModule(sc)
+        demuxer = demuxer_lib.ConfiguratorDemuxer()
+        rpc_mgr = cfgr.ConfiguratorRpcManager(sc, cm, conf, demuxer)
+        return sc, conf, rpc_mgr
+
+    def _get_GenericConfigRpcManager_object(self, conf, sc):
+        agent = gc.GenericConfigRpcManager(sc, conf)
+        return agent, sc
+
+    @mock.patch(__name__ + '.FakeObjects.nqueue')
+    @mock.patch(__name__ + '.FakeObjects.drivers')
+    def _get_GenericConfigEventHandler_object(self, sc, rpcmgr, drivers, nqueue):
+        agent = gc.GenericConfigEventHandler(sc, drivers, rpcmgr, nqueue)
+        return agent
+
+    def _get_FWaasRpcManager_object(self, conf, sc):
+        agent = fw.FWaasRpcManager(sc, conf)
+        return agent, sc
+
+    def _test_generic_event_creation(self, operation, method, bulk=False):
+        sc, conf, rpc_mgr = self._get_ConfiguratorRpcManager_object()
+        agent, sc = self._get_GenericConfigRpcManager_object(conf, sc)
+        
+        if bulk:
+            request_data = self.fo.fake_request_data_generic_bulk()
+            request_data1 = self.fo.fake_request_data_generic_bulk()
+        else:
+            request_data = self.fo.fake_request_data_generic_single()
+            request_data1 = self.fo.fake_request_data_generic_single()
+
+        if (method == 'CONFIGURE_ROUTES' or method == 'CLEAR_ROUTES'):
+            request_data1['config'][0]['resource'] = 'routes'
+            request_data['config'][0]['resource'] = 'routes'
+
+        with mock.patch.object(
+                    sc, 'new_event', return_value='foo') as mock_sc_event, \
+            mock.patch.object(sc, 'post_event') as mock_sc_rpc_event, \
+            mock.patch.object(rpc_mgr,
+                              '_get_service_agent_instance',
```

Comment:
Because service agents will be loaded only when the configurator runs.

Regards,
Deepak

On Thu, Mar 10, 2016 at 5:00 PM, njagadish <notifications@github.com> wrote:

> In gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py
> <https://github.com/oneconvergence/group-based-policy/pull/80#discussion_r55666526>
> :
>
> > +        if bulk:
> > +            request_data = self.fo.fake_request_data_generic_bulk()
> > +            request_data1 = self.fo.fake_request_data_generic_bulk()
> > +        else:
> > +            request_data = self.fo.fake_request_data_generic_single()
> > +            request_data1 = self.fo.fake_request_data_generic_single()
> > +
> > +        if (method == 'CONFIGURE_ROUTES' or method == 'CLEAR_ROUTES'):
> > +            request_data1['config'][0]['resource'] = 'routes'
> > +            request_data['config'][0]['resource'] = 'routes'
> > +
> > +        with mock.patch.object(
> > +                    sc, 'new_event', return_value='foo') as mock_sc_event, \
> > +            mock.patch.object(sc, 'post_event') as mock_sc_rpc_event, \
> > +            mock.patch.object(rpc_mgr,
> > +                              '_get_service_agent_instance',
>
> Why this function needs to be mocked? This is internal function of RPC
> manager and should be tested correct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oneconvergence/group-based-policy/pull/80/files#r55666526>
> .
>